### ISSUE TITLE
virtual marble mouse scroll wheel configured with sway no longer works in gamescope sessions

### ISSUE THREAD

**Issue by kevenwyld** [ValveSoftware/gamescope#1295]
Bear with me, this is a bit of a niche use case but does appear to be an actual bug somewhere...

Up until recently everything was fine however as of about 2-3 weeks ago one of my mice won't scroll in gamescope sessions.  Everything else works though, even the games in question if I don't use gamescope.

I have configured my Logitech Marble Mouse to have a virtual scroll wheel when holding down the "down" or "back" button.  This is done with libinput configuration in sway as follows (per the wiki):

```
input "1133:50184:Logitech_USB_Trackball" {
    scroll_method on_button_down
}
```

This works flawlessly in every application I've ever used, and up until recently it worked fine in gamescope too.  Now scrolling simply does nothing.  The rest of the mouse, and the back/forward buttons, work just fine.  It's only scrolling which is an issue.  Other mice with normal/default wheels scroll fine in gamescope.

I tried running xev inside a gamescope session as a simple test but as you can imagine it simply doesn't receive input from the marble mouse when scrolling, so there are no meaningful logs there.  It does receive input from other mice when scrolling (button 5 and 4).  It does provide an easy way to reproduce the problem though, you don't even need steam or a game.

xev when run outside of a gamescope session receives button 5 and 4 from both the marble mouse and other mice just fine.

I've tried both the stable version from arch (3.14.3) and building from master branch, there is no difference.

I will continue to try to debug, as I'm sure this is very low priority for everyone.  There can't be that many people in the world who use sway+gamescope+marblemouse .  I'm mostly posting here for visibility in case there happens to be even one other user of this combination of devices....

**Comment by Joshua-Ashton** [comment]
Probably worth checking if the input thread in wayland_backend is getting any scroll info at all.

**Comment by kevenwyld** [comment]
@Joshua-Ashton sorry I'm afraid I'm not sure where to start with that.  Could you point me in the right direction?  I did try `--debug-events` (probably not what you are talking about) which seems to report various types of device input however it doesn't report any scroll input.

**Comment by craigers521** [comment]
@kevenwyld you ever sort this out? I was having an issue with lutris where if I alt-tabbed or switched focus to a different app the game video would freeze. Turning on gamescope fixed that issue but now mouse scroll is no longer working. Works fine if I turn gamescope off but im a serial app switcher lol.  I'm on bazzite gnome, using latest wine.

**Comment by kevenwyld** [comment]
> @kevenwyld you ever sort this out? I was having an issue with lutris where if I alt-tabbed or switched focus to a different app the game video would freeze. Turning on gamescope fixed that issue but now mouse scroll is no longer working. Works fine if I turn gamescope off but im a serial app switcher lol. I'm on bazzite gnome, using latest wine.

I didn't, I've sortof just been dealing with it.  Do you have the same marble mouse and does scrolling work with other mice just not that one?  My scrolling issue is limited to the marble mouse specifically, others work fine.

**Comment by craigers521** [comment]
Yeah just a normal Logitech mouse. No idea why scrolling specifically
doesn't work in gamescope

**Comment by kevenwyld** [comment]
So I stumbled across a workaround for this completely by accident while trying to deal with a different problem.  Using the `--backend sdl` option for gamescope allows me to scroll with my marble mouse.

I have no clue why, although I have a wild guess:

It could be that the default on sway (wayland) is `--backend wayland` and that using sdl interprets input devices with SDL as well.  So this could be a problem with sway not passing scroll events to whatever type of window the normal gamescope session is for my particular mouse.

Anyway since this is such a niche issue impacting probably only me, since I might be the only sway user on earth that plays games with a marble mouse, it's likely I'll be stuck with this workaround for the foreseeable future.
